design/one-pager-managed-resource-api-design.md

```diff
@@ -0,0 +1,294 @@
+  sub-resource of the managed resource because this is where the most up-to-date information about the resource is
+  stored and we are able to limit RBAC of these users to access only `Status`.
+  
+### Naming Conventions
+
+There are two naming decisions here to make:
+* Managed resource CR name
+* Identifier string of the external resource in the cloud provider.
+
+_Name_ in this context means the identifier string for the resource in its environment. CR name is the identifier
+for resource in Kubernetes environment. But for external resources, this can be either `id`, `uid` or `name` properties
+of the resource. The one we want to name is the one that is used when _referring to_ that resource in the same
+environment. For example, in AWS, VPCs do not have a name but `VpcID` properties that is used for identification.
+
+#### Custom Resource Instance Name
+
+The name of the managed resource CR, which lives in our control cluster. The claim controller who creates the managed
+resource gives its name or some other entity might create the managed resource, like user themselves. So, do not assume
+that all the managed resources will have the same naming format anywhere.
+One enforcing condition is that it has to be unique amongst other instances of the CRD. The best way to guarantee that
+is to include Kubernetes UID in the name. The following format is valid as long as the length of the name is not longer
+than 253 characters. Given that we know UID is 36 characters long, name for the service you choose should be
+differentiable enough with 217 characters.
+
+For example, the following would be a good name for a CloudSQL managed resource CR:
+```
+cloudsql-7cb46d78-4556-4904-93d2-b34f6d5ccadf
+```
+
+#### External Resource Name
+
+Related to https://github.com/crossplaneio/crossplane/issues/624
+
+The decision for an external resource name is made by the controller of that managed resource in `Create` phase, note
+that this is different than CR name which claim controller or user decides on. In this case, we still have the
+possibility that the user themselves might want to name the external resource. This is why managed resource controller
+should only choose a name and go for it only if the respective field in `Spec` is empty. If it's not, we don't override
+it. The `Create` call uses whatever it sees in the `Spec`
+
+There are cases where naming by the API caller is not allowed, like VPC in AWS, we do not make any decision and do not
+include a field for that in the `Spec` since it's not configurable by the user. We include it in the `Status` when we
+receive it and other entities access `Status` to fetch the name.
+
+If we are able to provide a name, the decision flow is as following:
+
+1. Use CR name if the provider allows 253 characters for resource name. If not:
+2. Write your own name generation format with a random string generator that you can be sure the generated name
```

Comment:
What if:
* the provider auto-generates unique names
* the provider lets you supply a name
* the character max is less than 253

I think this may be a common case worth included in this ruleset.

Reply:
How about this; as long as `resourceType+UID` is less than what's allowed by the provider, use it. If it's longer that allowed, do not cut the tail, implement your own random generator for UID part whose length is suitable.